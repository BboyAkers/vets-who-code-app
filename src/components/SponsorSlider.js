import React from 'react'
import Slider from 'react-slick'
import FluidImage from './FluidImage'

const settings = {
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 8,
  className: 'sponsor-slider',
  centerMode: true,
  centerPadding: 30,
  slidesToScroll: 1,
  autoplay: true,
  autoplaySpeed: 3000,
  responsive: [
    {
      breakpoint: 990,
      settings: {
        slidesToShow: 4,
        slidesToScroll: 1,
      },
    },
    {
      breakpoint: 480,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 1,
      },
    },
  ],
}

function SponsorSlider() {
  return (
    <Slider {...settings}>
      <div>
        <a href="https://www.google.com/" target="_blank" rel="noopener noreferrer">
          <FluidImage fileName="google.png" alt="google" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://github.com/" target="_blank" rel="noopener noreferrer">
          <FluidImage fileName="github.png" alt="Github" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://repl.it" target="_blank" rel="noopener noreferrer">
          <FluidImage fileName="repl.it.png" alt="Repl.it" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://slack.com/" target="_blank" rel="noopener noreferrer">
          <FluidImage fileName="slack.png" alt="Slack" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://www.digitalocean.com/" target="_blank" rel="noopener noreferrer">
          <FluidImage fileName="DO.png" alt="Digital Ocean" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://frontendmasters.com/" target="_blank" rel="noopener noreferrer">
          <FluidImage
            fileName="fem.png"
            alt="Front End Masters"
            style={{ height: 60, width: 60 }}
          />
        </a>
      </div>
      <div>
        <a
          href="https://corporate.comcast.com/company/nbcuniversal"
          target="_blank"
          rel="noopener noreferrer"
        >
          <FluidImage fileName="comcast.png" alt="Comcast" style={{ height: 60, width: 60 }} />
        </a>
      </div>
      <div>
        <a href="https://www.contentful.com/" rel="noopener noreferrer" target="_blank">
          <FluidImage
            fileName="contentful.png"
            alt="Powered by Contentful"
            style={{ height: 60, width: 60 }}
          />
        </a>
      </div>
    </Slider>
  )
}

export default SponsorSlider
