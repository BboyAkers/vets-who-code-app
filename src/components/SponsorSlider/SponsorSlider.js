import React, { useState, useEffect, useContext } from 'react'
import FluidImage from '../FluidImage'
import Carousel from 'nuka-carousel'
import { ThemeContext } from '../../store/ThemeProvider'
import { FaSlack, FaGoogle, FaGithub } from 'react-icons/fa'
import { SiReplDotIt } from 'react-icons/si'
import { Contentful, Fem } from '../../icons'

const baseSettlings = {
  autoplay: true,
  enableKeyboardControls: true,
  pauseOnHover: true,
  speed: 500,
  wrapAround: true,
  withoutControls: true,
}

const alignmentStyles = {
  height: 60,
  width: 60,
  margin: '0 auto',
}

const elements = [
  {
    href: 'https://www.google.com/',
    label: 'Link to Google',
    title: 'Google',
    lightElement: 'google.png',
    darkElement: <FaGoogle size={60} color="#ffffff" />,
  },
  {
    href: 'https://github.com/',
    label: 'Link to Github',
    title: 'GitHub',
    lightElement: 'github.png',
    darkElement: <FaGithub size={60} color="#ffffff" />,
  },
  {
    href: 'https://repl.it',
    label: 'Link to Replit ',
    title: 'Repl.it',
    lightElement: 'repl.it.png',
    darkElement: <SiReplDotIt size={60} color="#ffffff" />,
  },
  {
    href: 'https://slack.com/',
    label: 'Link to Slack',
    title: 'Slack',
    lightElement: 'slack.png',
    darkElement: <FaSlack size={60} color="#ffffff" />,
  },
  {
    href: 'https://frontendmasters.com/',
    label: 'Link to Front End Masters',
    title: 'Front End Masters',
    lightElement: 'fem.png',
    darkElement: <Fem size={`60`} color="#ffffff" />,
  },
  {
    href: 'https://www.contentful.com/',
    label: 'Link to Contentful',
    title: 'Contentful',
    lightElement: 'contentful.png',
    darkElement: <Contentful size={`60`} color="#091f40" />,
  },
]

function SponsorSlider() {
  const { colorMode } = useContext(ThemeContext)

  const [viewport, setViewport] = useState(800)

  function updateWindowDimensions() {
    setViewport(window.innerWidth)
  }

  useEffect(() => {
    window.addEventListener('resize', updateWindowDimensions)
    //const root = window.document.documentElement
    //const initialColorMode = root.getAttribute('color-mode')
    //setColorMode(initialColorMode)
    return () => window.removeEventListener('resize', updateWindowDimensions)
  }, [])

  const isMobile = Boolean(viewport < 800)

  const lightElements = () => {
    return elements.map(data => {
      return (
        <>
          <a
            key={data.title}
            href={data.href}
            aria-label={data.label}
            title={data.title}
            target="_blank"
            rel="noopener noreferrer"
          >
            <FluidImage fileName={data.lightElement} alt={data.title} style={alignmentStyles} />
          </a>
        </>
      )
    })
  }

  const darkElements = () => {
    return elements.map(data => {
      return (
        <>
          <div key={data.title} style={{ textAlign: 'center' }}>
            <a
              href={data.href}
              aria-label={data.label}
              title={data.title}
              target="_blank"
              rel="noopener noreferrer"
            >
              {data.darkElement}
            </a>
          </div>
        </>
      )
    })
  }

  return (
    <Carousel
      {...baseSettlings}
      slidesToShow={isMobile ? 4 : 6}
      transitionMode={isMobile ? 'scroll' : 'fade'}
    >
      {colorMode === 'light' ? lightElements() : darkElements()}
    </Carousel>
  )
}

export default SponsorSlider
