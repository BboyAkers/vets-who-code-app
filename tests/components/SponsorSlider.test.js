import React from 'react'
import SponsorSlider from '../../src/components/SponsorSlider'

describe('<SponsorSlider />', () => {
  let wrapper

  beforeEach(() => wrapper = shallow(<SponsorSlider />))
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
