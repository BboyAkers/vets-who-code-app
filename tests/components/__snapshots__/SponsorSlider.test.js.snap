// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SponsorSlider /> should render correctly 1`] = `
<Slider
  autoplay={true}
  autoplaySpeed={3000}
  centerMode={true}
  centerPadding={0}
  className="slider"
  dots={true}
  infinite={true}
  responsive={
    Array [
      Object {
        "breakpoint": 990,
        "settings": Object {
          "slidesToScroll": 1,
          "slidesToShow": 4,
        },
      },
      Object {
        "breakpoint": 480,
        "settings": Object {
          "slidesToScroll": 1,
          "slidesToShow": 3,
        },
      },
    ]
  }
  slidesToScroll={1}
  slidesToShow={7}
  speed={500}
>
  <div>
    <a
      href="https://www.pluralsight.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="google"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://github.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="GitHub"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://repl.it"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="repl.it"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://slack.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Slack"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://www.digitalocean.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Digital Ocean"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://frontendmasters.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Front End Masters"
        height="60"
        src="image-mock"
        width="60"
      />
    </a>
  </div>
  <div>
    <a
      href="https://corporate.comcast.com/company/nbcuniversal/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Comcast"
        height="60"
        src="image-mock"
        width="80"
      />
    </a>
  </div>
</Slider>
`;
